### ISSUE TITLE
Constructor functions for Show to use receiver function

### ISSUE THREAD

**Issue by pakoito** [arrow-kt/arrow#830]
From gitter, current encoding

```
val catShow = Show<Cat> {
        val name = it.name
        val age = it.age
        val color = it.color
        "$name is a $age year old $color cat"
    }
```

with proposed encoding

```
val catShow = Show<Cat> {
        "$name is a $age year old $color cat"
    }
```

**Comment by pakoito** [comment]
`Show` cannot have receivers for two parameters, silly me :P

**Comment by ersin-ertan** [comment]
> <s>`Show`</s> `Eq` cannot ...

**Comment by pakoito** [comment]
Closed by #835 . Thanks @ersin-ertan
